src/merkle.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright 2018 int08h LLC
```

Comment:
Hmm, I'm not sure what to do here w.r.t. copyright. I think you retain that (or can if you wish). 

As long as the code is Apache 2.0 licensed I'm good with which way you want to go.

Reply:
I have no real opinion on copyright headers - I can change it to whatever makes things simpler for you.